File: Ejemplo2.cpp
#include <iostream>
using namespace std;
#include <list>
void Cambiar(int arreglo[]){
    arreglo[0] = 20;
}

int main(){
    int x[2] = {1,2};
    Cambiar(x);
    cout << x[3] << endl;

    //int *p = x;
    //p = p + 1;
    //cout << "Valor de p : " << p[0] << endl;
    //cout << "Valor de p : " << *p << endl;
    return 0;
}

File: Ejemplo2.java
public class Ejemplo2{
    private static void Cambiar(int arreglo[]){
        arreglo[0] = 20;
    }
    public static void main(String args[]){
        int x[] = {1,2};
        Cambiar(x);
        System.out.println(x[0]);        
    }
}